MaxText/layers/quantizations.py

```diff
@@ -60,6 +63,10 @@ def dot_general_cls(self, mesh_axes: Tuple[str, ...] = ()):
     """Placeholder for dot_general implementation in subclasses."""
     pass
 
+  def einsum(self, dtype: DType = jnp.float32):
```

Comment:
we'll prob want to generify the arguments for this abstract method later, but this is good for now!